runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -330,6 +330,10 @@ public UUID getStreamID() {
     @Deprecated // TODO: Add replacement method that conforms to style
     @SuppressWarnings({"checkstyle:membername", "checkstyle:abbreviation"}) // Due to deprecation
     private <R> R TXExecuteInner(Supplier<R> txFunction, boolean isMetricsEnabled) {
+        // This should not nest transactions
```

Comment:
Maybe a better comment would be, "don't nest transactions if we are already running transactionally"